filters/filter.go

```diff
@@ -22,103 +22,171 @@ package filters
 
 import (
 	"bytes"
+	"errors"
 	"fmt"
-	"sort"
+	"strings"
 )
 
-// logicalOp is a logical operator
-type logicalOp string
+var (
+	errInvalidFilterPattern = errors.New("invalid filter pattern defined")
+)
+
+// LogicalOp is a logical operator
+type LogicalOp string
```

Comment:
It's used inside rules and will be used when creating other combination filters moving forward.